docs/design/datacontracts/data_descriptor.md

```diff
@@ -0,0 +1,264 @@
+blob will augment the baseline with a known offset for all fields in the baseline.
+
+The `VALUE` may be a JSON numeric constant integer or a string containing a signed or unsigned
+decimal or hex (with prefix `0x` or `0X`) integer constant.  The constant must be within the range
+of the type of the global value.
+
+For pointer and nuint globals, the value may be assumed to fit in a 64-bit unsigned integer.  For
+nint globals, the value may be assumed to fit in a 64-bit signed integer.
+
+If the value is given as `{"indirect": int}` then the value is stored in an auxiliary array that is
+part of the data contrat descriptor.  Only in-memory data descriptors may have indirect values; baseline data descriptors may not have indirect values.
+
+Rationale: This allows tooling to generate the in-memory data descriptor as a single constant
+string.  For pointers, the address can be stored at a known offset in an in-proc
+array of pointers and the offset written into the constant json string.
+
+The indirection array is not part of the data descriptor spec.  It is expected that the data
+contract descriptor will include it. (The data contract descriptor must contain: the data
+descriptor, the set of compatible algorithmic contracts, the aux array of globals).
+
+## Example
+
+This is an example of a baseline descriptor for a 64-bit architecture. Suppose it has the name `"example-64"`
+
+```jsonc
+{
+  "version": 0,
+  "types": [
+    {
+      "name": "GCHandle",
+      "size": 8,
+      "fields": [
+        { "name": "Value", "type": "pointer", "offset": 0 }
+      ]
+    },
+    {
+      "name": "Thread",
+      "size": "indeterminate",
+      "fields": [
+        { "name": "ThreadId", "type": "uint32", "offset": "unknown" },
+        { "name": "Next", "type": "pointer" }, // offset "unknown" is implied
+        { "name": "ThreadState", "type": "uint32" }
+      ]
+    },
+    {
+      "name": "ThreadStore",
+      "fields": [
+        { "name": "ThreadCount", "type": "int32" },
+        { "name": "ThreadList", "type": "pointer" }
+      ]
+    }
+  ],
+  "globals": [
+    { "name": "FEATURE_EH_FUNCLETS", "type": "uint8", "value": "0" }, // baseline defaults value to 0
+    { "name": "s_pThreadStore", "type": "pointer" } // no baseline value
+  ]
+}
+```
+
+The following is an example of an in-memory descriptor that references the above baseline:
+
+```jsonc
+{
+  "version": "0",
+  "baseline": "example-64",
+  "types": [
+    {
+      "name": "Thread",
+      "fields": [
+        { "name": "ThreadId", "offset": 32 },
+        { "name": "ThreadState", "offset": 0 },
+        { "name": "Next", "offset": 128 }
+      ]
+    },
+    {
+      "name": "ThreadStore",
+      "fields": [
+        { "name": "ThreadCount", "offset": 32 }
+        { "name": "ThreadList", "offset": 8 }
+      ]
+    }
+  ],
+  "globals": [
+    { "name": "s_pThreadStore", "value": { "indirect": 0 } }
```

Comment:
Should `indirect` be 1 in this example? The value at index 0 below is `FEATURE_EH_FUNCLETS`.

Reply:
No.  the value of FEATURE_EH_FUNCLETS in this example is stored in the baseline and the in-memory version happened to match and doesn't override it.  (Even if it did override it, we would probably store it as a literal value in the json since it is known at build time and won't vary at execution time.  There is no need to store constant values in the aux array).  I'll clarify the example